test/suites/integration/scheduling_test.go

```diff
@@ -70,353 +72,556 @@ var _ = Describe("Scheduling", Ordered, ContinueOnFailure, func() {
+			}, v1.Container{
+				RestartPolicy: lo.ToPtr(v1.ContainerRestartPolicyAlways),
+				Resources: v1.ResourceRequirements{
+					Requests: v1.ResourceList{v1.ResourceCPU: resource.MustParse("500m")},
+				},
+			}),
+			Entry("container requirements < init container requirements", "2", v1.ResourceRequirements{
+				Requests: v1.ResourceList{v1.ResourceCPU: resource.MustParse("500m")},
+			}, ephemeralInitContainer(v1.ResourceRequirements{
+				Requests: v1.ResourceList{v1.ResourceCPU: resource.MustParse("250m")},
+			}), ephemeralInitContainer(v1.ResourceRequirements{
+				Requests: v1.ResourceList{v1.ResourceCPU: resource.MustParse("1250m")},
+			})),
+			Entry("container requirements > init container requirements", "1", v1.ResourceRequirements{
+				Requests: v1.ResourceList{v1.ResourceCPU: resource.MustParse("500m")},
+			}, ephemeralInitContainer(v1.ResourceRequirements{
+				Requests: v1.ResourceList{v1.ResourceCPU: resource.MustParse("300m")},
+			}), ephemeralInitContainer(v1.ResourceRequirements{
+				Requests: v1.ResourceList{v1.ResourceCPU: resource.MustParse("300m")},
+			})),
+			Entry("container requirements + sidecar requirements < init container requirements", "2", v1.ResourceRequirements{
+				Requests: v1.ResourceList{v1.ResourceCPU: resource.MustParse("250m")},
+			}, ephemeralInitContainer(v1.ResourceRequirements{
+				Requests: v1.ResourceList{v1.ResourceCPU: resource.MustParse("1250m")},
+			}), v1.Container{
+				RestartPolicy: lo.ToPtr(v1.ContainerRestartPolicyAlways),
+				Resources: v1.ResourceRequirements{
+					Requests: v1.ResourceList{v1.ResourceCPU: resource.MustParse("250m")},
+				},
+			}),
+			Entry("container requirements + sidecar requirements > init container requirements", "2", v1.ResourceRequirements{
+				Requests: v1.ResourceList{v1.ResourceCPU: resource.MustParse("750m")},
+			}, ephemeralInitContainer(v1.ResourceRequirements{
+				Requests: v1.ResourceList{v1.ResourceCPU: resource.MustParse("750m")},
+			}), v1.Container{
+				RestartPolicy: lo.ToPtr(v1.ContainerRestartPolicyAlways),
+				Resources: v1.ResourceRequirements{
+					Requests: v1.ResourceList{v1.ResourceCPU: resource.MustParse("750m")},
+				},
+			}),
+			Entry("sidecar requirements + later init requirements do not exceed container requirements", "1", v1.ResourceRequirements{
+				Requests: v1.ResourceList{v1.ResourceCPU: resource.MustParse("400m")},
+			}, ephemeralInitContainer(v1.ResourceRequirements{
+				Requests: v1.ResourceList{v1.ResourceCPU: resource.MustParse("300m")},
+			}), v1.Container{
+				RestartPolicy: lo.ToPtr(v1.ContainerRestartPolicyAlways),
+				Resources: v1.ResourceRequirements{
+					Requests: v1.ResourceList{v1.ResourceCPU: resource.MustParse("350m")},
+				},
+			}, ephemeralInitContainer(v1.ResourceRequirements{
+				Requests: v1.ResourceList{v1.ResourceCPU: resource.MustParse("300m")},
+			})),
+		)
+		DescribeTable(
+			"should provision a right-sized node when a pod has InitContainers (memory)",
```

Comment:
Do we need both CPU and memory testing? I'd assume that functional testing should catch that we are getting this wrong with different resources? We aren't testing with ephemeral storage and the cost of running additional testing in E2E is going to get more expensive over time

Reply:
Probably not no, I think it is valuable to include at least one test with multiple resource types but I think we can remove the memory `DescribeTable`.